exec/java-exec/src/main/resources/rest/static/js/ace-code-editor/mode-sql.js

```diff
@@ -1 +1,128 @@
\ No newline at end of file
+/**
+ * Drill SQL Definition (Forked from SqlServer definition)
+ */
+
+ace.define("ace/mode/sql_highlight_rules",["require","exports","module","ace/lib/oop","ace/mode/text_highlight_rules"], function(require, exports, module) {
+"use strict";
+
+var oop = require("../lib/oop");
+var TextHighlightRules = require("./text_highlight_rules").TextHighlightRules;
+
+var SqlHighlightRules = function() {
+
+    //TODO: https://drill.apache.org/docs/reserved-keywords/
+	//e.g. Cubing operators like ROLLUP are not listed
+    var keywords = (
+        "select|insert|update|delete|from|where|and|or|group|by|order|limit|offset|having|as|case|" +
+        "when|else|end|type|left|right|join|on|outer|desc|asc|union|create|table|key|if|" +
+        "not|default|null|inner|database|drop|" +
+		"flatten|kvgen|columns"
+    );
+	//Confirmed to be UnSupported as of Drill-1.12.0
+	/* cross|natural|primary|foreign|references|grant */
+
+    var builtinConstants = (
+        "true|false"
+    );
+
+    //Drill-specific
+    var builtinFunctions = (
+        //Math and Trignometric
+        "abs|cbrt|ceil|ceiling|degrees|e|exp|floor|log|log|log10|lshift|mod|negative|pi|pow|radians|rand|round|round|rshift|sign|sqrt|trunc|trunc|" +
```

Comment:
I believe n future we'll have system table with Drill functions. Will be there a possibility to pass list of functions to the js script instead of hard-coding them?

Reply:
I tried looking at that option, but it is quite difficult to figure out how the libraries read these files. We could do a dynamic generation of this list, when the {sys.functions} table does get implemented.

Reply:
Sounds good, could you please then just file a Jira for this improvement?